vulnerabilities/pipelines/v2_improvers/compute_advisory_todo.py

```diff
@@ -0,0 +1,347 @@
+            break
+        if not has_affected_package and impacted.affecting_vers:
+            has_affected_package = True
+        if not has_fixed_package and impacted.fixed_vers:
+            has_fixed_package = True
+
+    if has_affected_package and has_fixed_package:
+        return
+
+    if not has_affected_package and not has_fixed_package:
+        issue_type = "MISSING_AFFECTED_AND_FIXED_BY_PACKAGES"
+    elif not has_affected_package:
+        issue_type = "MISSING_AFFECTED_PACKAGE"
+    elif not has_fixed_package:
+        issue_type = "MISSING_FIXED_BY_PACKAGE"
+
+    if issue_type:
+        todo = AdvisoryToDoV2(
+            related_advisories_id=todo_id,
+            issue_type=issue_type,
+        )
+        todo_to_create.append(todo)
+        advisory_relation_to_create[todo_id] = [advisory]
+
+
+def check_conflicting_affected_and_fixed_by_packages_for_alias(
+    advisories,
+    cve,
+    todo_to_create,
+    advisory_relation_to_create,
+):
+    """
+    Add appropriate AdvisoryToDo for conflicting affected/fixed packages.
+
+    Compute the comparison matrix for the given set of advisories. Iterate through each advisory
+    and compute and store fixed versions and affected versionrange for each advisory,
+    keyed by purl.
+
+    Use the matrix to determine conflicts in affected/fixed versions for each purl. If for any purl
+    there is more than one set of fixed versions or more than one set of affected versions,
+    it means the advisories have conflicting opinions on the fixed or affected packages.
+
+    Example of comparison matrix:
+        {
+            "pkg:npm/foo/bar": {
+                "affected": {
+                    Advisory1: frozenset(VersionRange1, VersionRange2),
+                    Advisory2: frozenset(...),
+                },
+                "fixed": {
+                    Advisory1: frozenset(Version1, Version2),
+                    Advisory2: frozenset(...),
+                },
+            },
+            "pkg:pypi/foobar": {
+                "affected": {
+                    Advisory1: frozenset(...),
+                    Advisory2: frozenset(...),
+                },
+                "fixed": {
+                    Advisory1: frozenset(...),
+                    Advisory2: frozenset(...),
+                },
+            },
+            ...
+        }
+    """
+    matrix = {}
+    for advisory in advisories:
+        advisory_id = advisory.unique_content_id
+        for impacted in advisory.impacted_packages.all() or []:
+            if not impacted:
+                continue
+            affected_purl = str(impacted.base_purl)
+
+            initialize_sub_matrix(
+                matrix=matrix,
+                affected_purl=affected_purl,
+                advisory=advisory,
+            )
+
+            if fixed_version := impacted.fixed_vers:
+                matrix[affected_purl]["fixed"][advisory_id].add(str(fixed_version))
```

Comment:
Version range is already a string.
```suggestion
            if fixed_version_range := impacted.fixed_vers:
                matrix[affected_purl]["fixed"][advisory_id].add(fixed_version_range)
```